src/django_components/util/misc.py

```diff
@@ -79,8 +79,10 @@ def get_last_index(lst: List, key: Callable[[Any], bool]) -> Optional[int]:
     return None
 
 
-def _escape_js(js: str, eval: bool = True) -> str:
-    escaped_js = escape_js_string_literal(js)
+def _escape_js(content: str, wrap: bool = True, eval: bool = True) -> str:
```

Comment:
Are there lots of tests testing the security of this?

Reply:
No tests for that yet.

---

> Only thing I'm unsure about are the security aspects of this, the escaping/unescaping, are we sure they are bulletproof?

Good catch with the escaping. It's meant to escape the sequence `</script>`, which would break the script otherwise. Because user could define JS scripts like so:

```js
console.log("I'm am a <script></script> tag!");
```

Previously my approach was to wrap the JS script body in \` (backticks), and execute the script with `eval(unescape(body))`.

But I also found recently that the `eval()` function may be [blocked in stricter environments](https://stackoverflow.com/a/61823787/9788634), so ideally we'd avoid relying on it.

There's two places where there may be content inside `<script>` tag that could potentially contain `</script>` keyword:

1. When we insert the JS from `Component.js`
   - Checking with ChatGPT, I don't think there's a fully 100% reliable way to escape the sequence `</script>` from a JS script without breaking something.
     - The problem here is that we can't know if the `</script>` was used as part of a string or inside a comment or elsewhere. And to know that we'd have to parse the script as AST, and that's way too much effort. Because if we had these 3 lines:
       - `'I am </script>'`
       - `"I am </script>"`
       - `// I am </script>`

       Then the first two would need different ways of escaping them:
       - `'I am <' + '/script>'`
       - `"I am <" + "/script>"`
       - `// I am </script>`

   - So instead, I suggest that we simply check if the body in `Component.js` contains literal `</script>`. And if it does, we raise an error. That way this issue is captured before it gets to the browser.

2. When we insert the list of dependencies (JS / CSS) for the client-side dependency manager
   - The good thing here is that the dependencies are defined statically as JSON. But the json may still contain a literal `</script>`
   - But because the JSON is static and we don't need `eval()` to run it, I'm thinking of converting the entries inside the JSON to byte64, so any potential `</script>` are escaped.

---

So I'll fix update the escaping as outlined above and then I'll add some tests for that.

Reply:
I've updated this in https://github.com/EmilStenstrom/django-components/pull/845/commits/f8a7424ed6d343ff4a67234bae3f793775540a49. And added checks [here](https://github.com/EmilStenstrom/django-components/pull/845/commits/f8a7424ed6d343ff4a67234bae3f793775540a49#diff-a694855cccadd70fee2271f32ac0c207eef88cc9fcb252e1ee988773feec1c12R443).